virtualizarr/backend.py

```diff
@@ -133,9 +136,41 @@ def open_virtual_dataset(
         raise NotImplementedError()
 
     # if filetype is user defined, convert to FileType
+
     if filetype is not None:
         filetype = FileType(filetype)
 
+    if filetype == FileType.kerchunk:
+        # add parquet path
+        from fsspec.implementations.reference import LazyReferenceMapper
+
+        from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+        fs = _FsspecFSFromFilepath(filepath=filepath, reader_options=reader_options)
+
+        # yuck, but JSON doesn't have magic bytes
+        if fs.filepath.endswith(".json"):
+            import ujson
+
+            from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+            fpath = fs.open_file()
+
+            refs = ujson.load(fpath)
+            return dataset_from_kerchunk_refs(KerchunkStoreRefs(refs))
+
+        # also yuck, but the .parquet isn't actually a parquet it's a directory that contains named parquets for each group.
+        if fs.filepath.endswith("ref.parquet"):
+            lrm = LazyReferenceMapper(filepath, fs.fs)
+
+            # build reference dict from KV pairs in LazyReferenceMapper
+            # is there a better / more preformant way to extract this?
+            array_refs = {k: lrm[k] for k in lrm.keys()}
+
+            full_reference = {"refs": array_refs}
+
+            return dataset_from_kerchunk_refs(KerchunkStoreRefs(full_reference))
+
```

Comment:
what happens if the filetype is `"kerchunk"`, but neither the json nor the parquet check evaluate to true? Should we raise? As in, should the two `if`+`return` blocks be converted to `if`+`return`/`elif`+`return`/`else`+`raise`?

Reply:
Good call. I'll add an Exception.